### PR TITLE
fix(subscriptions): respect startingCSV

### PR DESCRIPTION
**Pull request by njhale**
Fixes `Subscription` resolution to respect the `StartingCSV` field.

Addresses [ALM-851](https://jira.coreos.com/browse/ALM-851).

### CODE REVIEW

File: pkg/controller/registry/resolver/querier_test.go [hunk 1]
```diff
@@ -239,7 +239,7 @@ func TestNamespaceSourceQuerier_FindPackage(t *testing.T) {
 			q := &NamespaceSourceQuerier{
 				sources: tt.fields.sources,
 			}
-			got, key, err := q.FindPackage(tt.args.pkgName, tt.args.channelName, tt.args.initialSource)
+			got, key, err := q.FindPackage(tt.args.pkgName, tt.args.channelName, "", tt.args.initialSource)
 			require.Equal(t, tt.out.bundle, got)
```
Comment (ecordell) [inline pkg/controller/registry/resolver/querier_test.go]: Unit tests for new querier behavior?

File: test/e2e/subscription_e2e_test.go [hunk 2]
```diff
@@ -489,3 +490,79 @@ func TestCreateNewSubscriptionManualApproval(t *testing.T) {
...
+	installPlanName := subscription.Status.Install.Name
+
+	// Wait for InstallPlan to be status: Complete before checking resource presence
+	fetchedInstallPlan, err := fetchInstallPlan(t, crc, installPlanName, buildInstallPlanPhaseCheckFunc(v1alpha1.InstallPlanPhaseComplete, v1alpha1.InstallPlanPhaseFailed))
+	require.NoError(t, err)
+	t.Logf("Install plan %s fetched with status %s", fetchedInstallPlan.GetName(), fetchedInstallPlan.Status.Phase)
+	require.Equal(t, v1alpha1.InstallPlanPhaseComplete, fetchedInstallPlan.Status.Phase)
+
+	// Wait for csvA to be found
+	fetchedCSV, err := fetchCSV(t, crc, csvA.GetName(), testNamespace, buildCSVConditionChecker(v1alpha1.CSVPhaseSucceeded, v1alpha1.CSVPhaseFailed))
```
Comment (ecordell) [inline test/e2e/subscription_e2e_test.go]: This could be racy, right? What if by the time this happens, we’ve synced again and updated from csvA to csvB?

Maybe we do an manual approval?

Comment (njhale) [reply test/e2e/subscription_e2e_test.go]: You're totally right - I just pushed up what I had locally. I think a manual approval might work.

File: pkg/controller/registry/resolver/querier_test.go (hunk 1 above)
Comment (njhale) [reply pkg/controller/registry/resolver/querier_test.go]: Will do.

File: pkg/controller/operators/catalog/operator.go [hunk 3]
```diff
@@ -530,14 +530,11 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 	// get the set of sources that should be used for resolution and best-effort get their connections working
 	logger.Debug("resolving sources")
 	resolverSources := o.ensureResolverSources(logger, namespace)
+	querier := resolver.NewNamespaceSourceQuerier(resolverSources)
 
 	logger.Debug("checking if subscriptions need update")
 
 	subs, err := o.lister.OperatorsV1alpha1().SubscriptionLister().Subscriptions(namespace).List(labels.Everything())
-	if err != nil {
```
[the remark this answers is not in the document]
Comment (njhale) [reply pkg/controller/operators/catalog/operator.go]: Good catch - probably happened during the rebase.

File: pkg/controller/registry/resolver/resolver.go [hunk 4]
```diff
@@ -112,7 +111,6 @@ func (r *OperatorsV1alpha1Resolver) ResolveSteps(namespace string, sourceQuerier
 		if subExists && existingSubscription.Status.CurrentCSV != op.Identifier() {
 			existingSubscription.Status.CurrentCSV = op.Identifier()
 			updatedSubs = append(updatedSubs, existingSubscription)
-			continue
```
Comment (ecordell) [inline pkg/controller/registry/resolver/resolver.go]: Why are these removed?

Comment (njhale) [reply pkg/controller/registry/resolver/resolver.go]: I didn't (and doesn't) look like this `continue` is skipping any logic between here and the next iteration of the loop.
